samples/tcp-echo/src/Dockerfile

```diff
@@ -16,7 +16,7 @@
 ARG BASE_DISTRIBUTION=default
 
 # build a tcp-echo binary using the golang container
-FROM golang:1.12 as builder
+FROM golang:1.14 as builder
```

Comment:
I'm not sure why you would pick the latest (currently 1.14.4) release of go here, but then explicitly use 1.14.2 in the build file.

Reply:
I got a reference from other Dockerfile in istio repo and assumed that we are not using the full version here. The reason to select `1.14.2` is because it's already tested in build tools https://github.com/istio/istio/issues/21567. However, I'm happy to change it to `1.14.2` instead of just `1.14`

Reply:
To me it's just a nit, but just wanting to call out that we do use different versions in different places, and as you found out it's sometimes hard to get them all changed when needed. 

Something to think about going forward. I know in some places we use GO_VERSION (or something close to that). Using that everywhere might help when searching for places to change, or maybe we could simply set the value in the Makefile and pass it to the various places.

Reply:
+1 for using `GO_VERSION`